packages/dd-trace/test/profiling/exporters/agent.spec.js

```diff
@@ -7,18 +7,39 @@ const os = require('os')
 const path = require('path')
 const getPort = require('get-port')
 const { gunzipSync } = require('zlib')
-const { perftools } = require('../../../../../protobuf/profile')
-const { Profile } = require('../../../src/profiling/profile')
+const CpuProfiler = require('../../../src/profiling/profilers/cpu')
+const HeapProfiler = require('../../../src/profiling/profilers/heap')
+const { perftools } = require('pprof/proto/profile')
+const semver = require('semver')
 
-const { decode, encode } = perftools.profiles.Profile
+if (!semver.satisfies(process.version, '>=10.12')) {
```

Comment:
This should either be removed, or the corresponding check should be re-added to the code.

Reply:
I think it would be worth it to also add a test for when `pprof` is not available and when the Node version is too low to make sure it doesn't result in crashing the entire tracer.

Reply:
The test uses the `pprof` profilers directly though. Even if the Profiler class itself protects against it, this test needs to skip too or it will crash when the `createProfile(...)` function tries to call `profiler.start(...)`.

Reply:
Not sure I understand. Are you saying the issue only exists in the test? How are we testing that the profiler is not also crashing in actual code for older versions?

Reply:
The Profiler class which manages each individual profiler and is responsible for calling start will not start any of the profilers if the Node.js version is less than v10. This specific test file, however, uses the cpu and heap profilers _directly_ so it doesn't have that escape hatch that the Profiler class would normally be applying, therefore that escape hatch needs to be duplicated here.